api/client/src/main/java/org/projectnessie/client/auth/oauth2/OAuth2Utils.java

```diff
@@ -30,4 +37,18 @@ static String randomAlphaNumString(int length) {
         .collect(StringBuilder::new, StringBuilder::appendCodePoint, StringBuilder::append)
         .toString();
   }
+
+  public static JsonNode fetchDiscoveryData(URI discoveryEndpoint, HttpClient httpClient) {
+    URI uri = new UriBuilder(discoveryEndpoint).path(".well-known/openid-configuration").build();
```

Comment:
Since the OIDC spec indicates that `.well-known/openid-configuration` is appended to the `issuer` URL. In that context using "discovery" in config names seems a bit misleading... WDYT about renaming the `authentication.oauth2.token-endpoint` config to `authentication.oauth2.issuer`? We can still talk about "discovery" in the javadoc / web docs, but I think using `issuer` as the config name is more precise.

Reply:
I agree that "discovery" is misleading, but the token endpoint is _not_ the issuer URL. I suggest instead:

* Use the term "Issuer URL" for what is currently named the "discovery endpoint";
* Rename `authentication.oauth2.discovery-endpoint` to `authentication.oauth2.issuer-url`;
* Keep using the terms "token endpoint" and "authorization endpoint" since these names adhere to standards.
* Also keep the existing options `authentication.oauth2.token-endpoint` and `authentication.oauth2.auth-endpoint`; they override whatever the discovery process finds out.

WDYT?

Reply:
That's exactly what I mean :)

Reply:
I mistyped in my original message :facepalm: I meant to rename the discovery config, of course.